src/components/KSlideout/KSlideout.vue

```diff
@@ -38,27 +93,96 @@
 </template>
 
 <script lang="ts" setup>
-import { onMounted, onUnmounted } from 'vue'
+import { onMounted, onUnmounted, computed, PropType } from 'vue'
 import KCard from '@/components/KCard/KCard.vue'
 import KIcon from '@/components/KIcon/KIcon.vue'
+import useUtilities from '@/composables/useUtilities'
+import type { BadgeAppearance, BadgeShape } from '@/types'
+import { BadgeAppearances, BadgeShapes } from '@/types'
 
 const props = defineProps({
   isVisible: {
     type: Boolean,
     default: false,
   },
+  // controls close button alignment
+  closeButtonAlignment: {
+    type: String,
+    default: 'start',
+    validator: (value: string): boolean => {
+      return ['start', 'end'].includes(value)
+    },
+  },
+  // enable/disable overlay to be able to interact with the background content while the slideout is expanded
+  overlayEnabled: {
+    type: Boolean,
+    default: true,
+  },
+  // allows a host app to define the offset from the top of the page
+  offsetTop: {
```

Comment:
Can we change this to take a string, so users can provide things like percentages or `vh`?

Reply:
TBH I don't think relative values make sense here -- you wouldn't offset the slideout based on `vh` or percentage for something like a navbar, etc.

Reply:
I don't think they would ever be used, so not sure support is needed.

e.g. if you have the top of the slideout set to a value of `10%` or `10vh` if the user resizes their browser window, the slideout will move from its position

Reply:
Okay, makes sense, I can revert this one.
@kaiarrowood - do you have a specific reason to support this prop as String instead of Number before I revert this change ?

Reply:
If you're going to force it as `numeric` I would just explicitly say in the docs that it takes a `number` and that `number` will use `px` as the unit. However you want to explain it so there is no confusion